docs/metrics.md

```diff
@@ -4,30 +4,158 @@ Let's look at the various metrics you can calculate using Valor.
+| mAP Averaged Over IOUs | The mean of several averaged AP scores, calculated over various classes. | $$\dfrac{1}{\text{number of thresholds}} \sum\limits_{iou \in thresholds} mAP_{iou}$$ |
 
 ## Semantic Segmentation Metrics
 
-| Name 	| Description 	|
-|---	|---	|
-| Intersection Over Union (IOU) 	| The overlap between the ground truth and predicted regions of an image, measured as a percentage, grouped by class. IOUs are calculated by a) fetching the ground truth and prediction rasters for a particular image and class, b) counting the true positive pixels (e.g., the number of pixels that were selected in both the ground truth masks and prediction masks), and c) dividing the sum of true positives by the total number of pixels in both the ground truth and prediction masks. |
-| Mean IOU 	| The average of IOUs, calculated over several different classes. 	|
+| Name | Description | Equation |
+| :- | :- | :- |
+| Intersection Over Union (IOU) | A ratio between the groundtruth and predicted regions of an image, measured as a percentage, grouped by class. |$$\dfrac{area( prediction \cap groundtruth )}{area( prediction \cup groundtruth )}$$ |
+| Mean IOU 	| The average of IOUs, calculated over several different classes. | $$\dfrac{1}{\text{number of classes}} \sum\limits_{c \in classes} IOU_{c}$$ |
+
+# Notes
+1. When calculating IOUs for object detection metrics, Valor handles the necessary conversion between different types of geometric annotations. For example, if your model prediction is a polygon and your groundtruth is a raster, then the raster will be converted to a polygon prior to calculating the IOU.
+
+# Appendix: Metric Calculations
+
+## Binary ROC AUC
+
+### Receiver Operating Characteristic (ROC)
+
+An ROC curve plots TPR vs. FPR at different confidence thresholds.
+
+See [Classification: ROC Curve and AUC](https://developers.google.com/machine-learning/crash-course/classification/roc-and-auc) for more information.
+
+#### Determining the rate of correct predictions.
+
+| Element | Description |
+| ------- | ------------ |
+| True Positive (TP) | Prediction confidence score >= threshold and is correct. |
+| False Positive (FP) | Prediction confidence score >= threshold and is incorrect. |
+| True Negative (TN) | Prediction confidence score < threshold and is correct. |
+| False Negative (FN) | Prediction confidence score < threshold and is incorrect. |
+
+- $\text{True Positive Rate (TPR)} = \dfrac{|TP|}{|TP| + |FN|}$
+
+- $\text{False Positive Rate (FPR)} = \dfrac{|FP|}{|FP| + |TN|}$
+
+These rates vary depending on a threshold so we can add a threshold parameter to be more explicit.
+
+$$
+TPR(threshold) = \dfrac{|TP(threshold)|}{|TP(threshold)| + |FN(threshold)|}
+$$
+
+$$
+FPR(threshold) = \dfrac{|FP(threshold)|}{|FP(threshold)| + |TN(threshold)|}
+$$
+
+In Valor, we use the confidence scores sorted in decreasing order as our thresholds.
```

Comment:
I was confused by what "threshold" meant in the above calculations until I read this sentence. maybe move this sentence up under the header, and change it to: 

"In Valor, we use the confidence scores sorted in decreasing order as our thresholds. Using these thresholds, we can calculate our TPR and FPR as follows:"